src/ahp/verifier.rs

```diff
@@ -138,4 +138,75 @@ impl<F: PrimeField> AHPForR1CS<F> {
 
         (query_set, state)
     }
+
+    /// Output the query state and next round state.
+    pub fn verifier_query_set<'a, 'b, R: RngCore>(
+        state: VerifierState<F>,
+        _: &'a mut R,
+    ) -> (QuerySet<'b, F>, VerifierState<F>) {
+
+        let beta = state.second_round_msg.unwrap().beta;
+        let gamma = state.gamma.unwrap();
+
+        let g_h = state.domain_h.group_gen();
+        let g_k = state.domain_k.group_gen();
+
+        let mut query_set = QuerySet::new();
+
+        // Inner sumcheck
+
+        // First round polys
+        query_set.insert(("w".into(), ("beta".into(), beta)));
+        query_set.insert(("z_a".into(), ("beta".into(), beta)));
+        query_set.insert(("z_b".into(), ("beta".into(), beta)));
+
+        // Second round polys
+        query_set.insert(("t".into(), ("beta".into(), beta)));
+        query_set.insert(("z_1".into(), ("beta".into(), beta)));
+        query_set.insert(("z_1".into(), ("g * beta".into(), g_h * beta)));
+        query_set.insert(("h_1".into(), ("beta".into(), beta)));
+
+        // Outer sumcheck
+
+        // Third round polys
+        query_set.insert(("z_2".into(), ("gamma".into(), gamma)));
+        query_set.insert(("z_2".into(), ("g * gamma".into(), g_k * gamma)));
+        query_set.insert(("h_2".into(), ("gamma".into(), gamma)));
+        query_set.insert(("a_row".into(), ("gamma".into(), gamma)));
+        query_set.insert(("a_col".into(), ("gamma".into(), gamma)));
+        query_set.insert(("a_row_col".into(), ("gamma".into(), gamma)));
+        query_set.insert(("a_val".into(), ("gamma".into(), gamma)));
+        query_set.insert(("b_row".into(), ("gamma".into(), gamma)));
+        query_set.insert(("b_col".into(), ("gamma".into(), gamma)));
+        query_set.insert(("b_row_col".into(), ("gamma".into(), gamma)));
+        query_set.insert(("b_val".into(), ("gamma".into(), gamma)));
+        query_set.insert(("c_row".into(), ("gamma".into(), gamma)));
+        query_set.insert(("c_col".into(), ("gamma".into(), gamma)));
+        query_set.insert(("c_row_col".into(), ("gamma".into(), gamma)));
+        query_set.insert(("c_val".into(), ("gamma".into(), gamma)));
+
+        (query_set, state)
+    }
+
+    /// Evaluate the given polynomials at `query_set` and returns a Vec<((poly_label, point_label), eval)>)
```

Comment:
Move this to poly commit please (the original "evaluate_query_set" function is also in poly-commit

Reply:
Done